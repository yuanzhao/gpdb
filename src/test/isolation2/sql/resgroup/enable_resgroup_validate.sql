-- start_ignore
! rmdir /sys/fs/cgroup/cpu/gpdb/*/;
! rmdir /sys/fs/cgroup/cpuacct/gpdb/*/;
! rmdir /sys/fs/cgroup/cpu/gpdb;
! rmdir /sys/fs/cgroup/cpuacct/gpdb;
-- end_ignore

-- gpdb top group is not created
! gpconfig -c gp_resource_manager -v group;

-- start_ignore
! mkdir /sys/fs/cgroup/cpu/gpdb;
! mkdir /sys/fs/cgroup/cpuacct/gpdb;
! chmod 644 /sys/fs/cgroup/cpu/gpdb;
-- end_ignore

-- gpdb directory should have rwx permission
! gpconfig -c gp_resource_manager -v group;

-- start_ignore
! chmod 755 /sys/fs/cgroup/cpu/gpdb;
! chmod 444 /sys/fs/cgroup/cpu/gpdb/cgroup.procs;
! chmod 444 /sys/fs/cgroup/cpu/gpdb/cpu.cfs_quota_us;
! chmod 244 /sys/fs/cgroup/cpu/gpdb/cpu.cfs_period_us;
! chmod 244 /sys/fs/cgroup/cpuacct/gpdb/cpuacct.usage;
-- end_ignore

-- cgroup.procs should have writepermission
-- cpu.cfs_quota_us should have write permission
-- cpu.cfs_period_us should have read permission
-- cpuacct.usage should have read permission
! gpconfig -c gp_resource_manager -v group;
