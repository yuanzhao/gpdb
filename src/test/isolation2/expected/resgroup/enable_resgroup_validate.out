-- start_ignore
! rmdir /sys/fs/cgroup/cpu/gpdb/*/;

! rmdir /sys/fs/cgroup/cpuacct/gpdb/*/;

! rmdir /sys/fs/cgroup/cpu/gpdb;

! rmdir /sys/fs/cgroup/cpuacct/gpdb;

-- end_ignore

-- gpdb top group is not created
! gpconfig -c gp_resource_manager -v group;
20170517:11:54:17:011348 gpconfig:nyu-vm-centos:gpadmin-[WARNING]:-Managing queries with resource groups is an experimental feature. A work-in-progress version is enabled.
20170517:11:54:18:011348 gpconfig:nyu-vm-centos:gpadmin-[CRITICAL]:-new GUC value failed validation: [nyu-vm-centos:cgroup is not properly configured: directory '/sys/fs/cgroup/cpu/gpdb/' does not exist
]
new GUC value failed validation: [nyu-vm-centos:cgroup is not properly configured: directory '/sys/fs/cgroup/cpu/gpdb/' does not exist
]


-- start_ignore
! mkdir /sys/fs/cgroup/cpu/gpdb;

! mkdir /sys/fs/cgroup/cpuacct/gpdb;

! chmod 644 /sys/fs/cgroup/cpu/gpdb;

-- end_ignore

-- gpdb directory should have rwx permission
! gpconfig -c gp_resource_manager -v group;
20170517:11:54:18:011409 gpconfig:nyu-vm-centos:gpadmin-[WARNING]:-Managing queries with resource groups is an experimental feature. A work-in-progress version is enabled.
20170517:11:54:18:011409 gpconfig:nyu-vm-centos:gpadmin-[CRITICAL]:-new GUC value failed validation: [nyu-vm-centos:cgroup is not properly configured: directory '/sys/fs/cgroup/cpu/gpdb/' permission denied: require permission 'rwx'
]
new GUC value failed validation: [nyu-vm-centos:cgroup is not properly configured: directory '/sys/fs/cgroup/cpu/gpdb/' permission denied: require permission 'rwx'
]


-- start_ignore
! chmod 755 /sys/fs/cgroup/cpu/gpdb;

! chmod 444 /sys/fs/cgroup/cpu/gpdb/cgroup.procs;

! chmod 444 /sys/fs/cgroup/cpu/gpdb/cpu.cfs_quota_us;

! chmod 244 /sys/fs/cgroup/cpu/gpdb/cpu.cfs_period_us;

! chmod 244 /sys/fs/cgroup/cpuacct/gpdb/cpuacct.usage;

-- end_ignore

-- cgroup.procs should have writepermission
-- cpu.cfs_quota_us should have write permission
-- cpu.cfs_period_us should have read permission
-- cpuacct.usage should have read permission
! gpconfig -c gp_resource_manager -v group;
20170517:11:54:18:011466 gpconfig:nyu-vm-centos:gpadmin-[WARNING]:-Managing queries with resource groups is an experimental feature. A work-in-progress version is enabled.
20170517:11:54:18:011466 gpconfig:nyu-vm-centos:gpadmin-[CRITICAL]:-new GUC value failed validation: [nyu-vm-centos:cgroup is not properly configured: file '/sys/fs/cgroup/cpu/gpdb/cgroup.procs' permission denied: require permission 'rw'
]
new GUC value failed validation: [nyu-vm-centos:cgroup is not properly configured: file '/sys/fs/cgroup/cpu/gpdb/cgroup.procs' permission denied: require permission 'rw'
]

